app/jobs/rstuf/check_job.rb

```diff
@@ -0,0 +1,18 @@
+class Rstuf::CheckJob < Rstuf::ApplicationJob
+  RetryException = Class.new(StandardError)
+  FailureException = Class.new(StandardError)
+  retry_on RetryException, wait: :exponentially_longer, attempts: 10
+
+  queue_with_priority PRIORITIES.fetch(:push)
+
+  def perform(task_id)
+    case Rstuf::Client.task_status(task_id)
+    when "SUCCESS"
+      # no-op, all good
+    when "FAILURE"
+      raise FailureException, "RSTUF job failed"
```

Comment:
How much do we care about this failing? Jobs fail often enough that these might be missed, right? Should we build a different way to record error percentage?

For example, should it make the request and ignore the result, then periodically check for consistency?

Reply:
I'm not 100% sure how much to "infect" RubyGems.org with RSTUF. Yes, it is possible to add `rstuf_indexed` to `Version` model and update in here.

Currently the main idea is to raise exception on problems. If RSTUF is not able to finish the task or task is failing, it would be great to get maintainers notified since it usually means something is wrong with RSTUF.

Reply:
Do we maintain the server side? If so, could we just error on the server?

Reply:
I'm OK with using persistent job failures as a signal for ONCALL to investigate, it's a pattern I've used before

Reply:
@martinemde plan is to maintain server side (deploy to RubyGems.org cluster). I'll see during deployment how simple that is to track errors in there. I'll definitelly work on improvements in there if needed. But I would like to keep RubyGems.org isolated and raise error if 3rd party app doesn't work. Maybe in the future RSTUF will be maintained by different maintainers.